advanced/multichain/rpc-reference/stacks-rpc.mdx

```diff
@@ -88,4 +88,43 @@ For signing a message with wallet users private key.
 }
 ```
 
-- `signature` - The signature of the message
\ No newline at end of file
+- `signature` - The signature of the message
+
+
+
+
+## stx_getAddresses
+
+For requesting the wallet addresses
+
+### Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "stx_getAddresses",
+}
+```
+
+### Response
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "accounts": [
+      { 
+        "address": "SP23....",
+        "publicKey": "ab12031..."
+      }
+    ]
+  }
+}
+```
+
+- `accounts`: The connected accounts with their address and publicKey
+
+
+### Session Properties
+In a connection request, it is recommended to serialize the response to `getAddresses` in `session.sessionProperties.stacks_getAddresses`. This allows dapps to consume an active session without requiring a context switch to re-request all addresses and associated public keys from the wallet.
```

Comment:
Should we add an example of how to do it in sessionProperties? It might not be super clear for devs CC @tomiir